src/ReactiveDomain.Identity/Domain/Client.cs

```diff
@@ -27,24 +38,33 @@ private void RegisterEvents()
                 Guid id,
                 Guid applicationId,
                 string clientName, //todo: value object?
-                string encryptedClientSecret)
+                string encryptedClientSecret,
+                string[] redirectUris,
+                string[] logoutRedirectUris,
+                string frontChannlLogoutUri,
+                ICorrelatedMessage source)
+                : base(source)
         {
             Ensure.NotEmptyGuid(id, nameof(id));
             Ensure.NotEmptyGuid(applicationId, nameof(applicationId));
             Ensure.NotNullOrEmpty(clientName, nameof(clientName));
             Ensure.NotNullOrEmpty(encryptedClientSecret, nameof(encryptedClientSecret));
+            Ensure.NotNullOrEmpty(redirectUris, nameof(frontChannlLogoutUri));
+            Ensure.NotNullOrEmpty(logoutRedirectUris, nameof(logoutRedirectUris));
+            Ensure.NotNullOrEmpty(frontChannlLogoutUri, nameof(redirectUris));
+            Ensure.NotNull(source, nameof(source));
             //todo: move url definitions into Elbe when adding the secret store
             Raise(new ClientCreated(
                   Id,
```

Comment:
Should be `id` since the `Id` property hasn't been set yet.

Reply:
fixed